ihatemoney/web.py

```diff
@@ -391,6 +396,83 @@ def edit_project():
     )
 
 
+@main.route("/<project_id>/upload_json", methods=["GET", "POST"])
+def upload_json():
+    form = UploadForm()
+    pid = g.project.id
+    if form.validate_on_submit():
+        filename = pid + "_uploaded_bills.json"
+        form.file.data.save(filename)
+        try:
+            import_project(filename)
```

Comment:
There we have the same issue where we create the file locally rather than in memory.

Reply:
By the way, since we already have the data available, it should be simpler to just parse them as JSON here, handles exceptions and give the JSON dict to `import_project`. Not touching the disk seems a good idea.